packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -905,6 +905,32 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
           'A ScrollController is required when Scrollbar.isAlwaysShown is true. '
           'Either Scrollbar.controller was not provided, or a PrimaryScrollController could not be found.',
         );
+        assert(
+          scrollController!.hasClients,
+          'The ScrollController has no positions attached. The Scrollbar cannot '
+          'be painted without a ScrollPosition.'
```

Comment:
👍 
This check can really help developers, such as #76783.
Can we also point out that the bar and scrollable widget need to use the same controller to help developers troubleshoot problems? It is common for the bar to use `PrimaryScrollController ` and widget to specify a new controller.

Reply:
Maybe this can also fix #75352, right?

Reply:
Yeah I think so, I am going to add a lot of documentation here as well, I am seeing some common patterns emerge from new issues that folks are filing.